interop/e2e_tests/postie_test.go

```diff
@@ -130,3 +130,165 @@ func TestPostieInboxRelay(t *testing.T) {
 	_, err = wait.ForReceiptOK(context.Background(), testSuite.OpSysA.Clients["sequencer"], tx.Hash())
 	require.NoError(t, err)
 }
+
+// This is testing whether a message can be replayed via the CrossL2Inbox
+// if the previous attempts resulted in a out of gas revert. Specifically, this test
+// calls CrossL2Inbox.RunCrossL2Message with not enough gas to trigger the
+// out of gas revert, then calls RunCrossL2Message with enough gas and
+// asserts success
+func TestPostieInboxReplay(t *testing.T) {
```

Comment:
Like finalized withdrawals for the OptimismPortal, we came to the conclusion that the inbox shouldn't allow for the replay of messages right?

Looking at this test, it looks like this isn't testing what you might have intended for `Safe.callWithMinGas` reverting with insufficient gas. This is reverting simply because the tx get's rejected from the EVM when executing due to low gas which is different

Reply:
I think we should change this test such that it doesn't allow for the replayability of a consumed message that has been reverted.

Perhaps also adding a few lines to `TestPostieInboxRelay` at the end that it also doesn't allow for the replayability of also a **successful** message as well

Reply:
That's right, should I refactor this test to trigger the `SafeCall: Not enough gas` revert and then assert that the message cannot be replayed? Is the test as is a useful assertion?

Reply:
If you wanted to trigger a `Safe.callWithMinGas` revert, i think you could probably invoke a message to the `InteropL2StandardBridge`. 

1. Measure how much Gas was needed to for the Inbox to relay just this ETH transfer
2. Send a message through the SB but use the Gas from (1). This should revert as all the contract code would need much more gas

Reply:
> it doesn't allow for the replayability of a consumed message that has been reverted.

Is that reverted because of [SafeCall: Not enough gas](https://github.com/ethereum-optimism/optimism/blob/610ad2ba602af7e9f70a6ff585e07f5ba6648123/packages/contracts-bedrock/src/libraries/SafeCall.sol#L121C32-L121C32), or a revert from the relayed `call` to the msg target? If it's the former, than the whole transaction is reverted and there's no issue replaying it with more gas. If it's that latter, than the message cannot be replayed and will revert with `CrossL2Inbox: message has already been consumed`

Reply:
reverted due to SafeCall: Not enough gas. I'm not 100% certain about the behavior of `SafeCall` if it can capture a revert of the call itself or the entire transaction reverts

Reply:
I don't think it is capturing the revert. Running the test locally results in a tx trace being logged, and the status of the tx is 0. I think the revert is only captured if the relayed call to the msg target reverts